router/usecase/worker/candidate_route_search_data_worker.go

```diff
@@ -0,0 +1,63 @@
+package worker
+
+import (
+	"context"
+
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	"github.com/osmosis-labs/sqs/log"
+	"github.com/osmosis-labs/sqs/sqsdomain"
+)
+
+type candidateRouteSearchDataWorker struct {
+	listeners                []domain.CandidateRouteSearchDataUpdateListener
+	poolsHandler             mvc.PoolHandler
+	candidateRouteDataHolder mvc.CandidateRouteSearchDataHolder
+	logger                   log.Logger
+}
+
+var (
+	_ domain.CandidateRouteSearchDataWorker = &candidateRouteSearchDataWorker{}
+	_ domain.PoolLiquidityComputeListener   = &candidateRouteSearchDataWorker{}
+)
+
+func NewCandidateRouteSearchDataWorker(poolHandler mvc.PoolHandler, candidateRouteDataHolder mvc.CandidateRouteSearchDataHolder, logger log.Logger) *candidateRouteSearchDataWorker {
+	return &candidateRouteSearchDataWorker{
+		listeners:                []domain.CandidateRouteSearchDataUpdateListener{},
+		poolsHandler:             poolHandler,
+		candidateRouteDataHolder: candidateRouteDataHolder,
+		logger:                   logger,
+	}
+}
+
+// OnPoolLiquidityCompute implements domain.PoolLiquidityComputeListener.
+func (c *candidateRouteSearchDataWorker) OnPoolLiquidityCompute(ctx context.Context, height uint64, blockPoolMetaData domain.BlockPoolMetadata) error {
+	// Compute search data and propagate error up the chain to fail the health check.
+	if err := c.ComputeSearchData(ctx, height, blockPoolMetaData); err != nil {
+		return err
+	}
+
+	// Notify listeners
+	for _, listener := range c.listeners {
+		_ = listener.OnSearchDataUpdate(ctx, height)
```

Comment:
Do we expect the listener will take long to finish? Might consider using separate thread.

Reply:
No, it is not expected. The only current use case is to have a health check run it.

I prefer introducing concurrency only when necessary for code simplicity, better debugging experience and bug resistance. 

This is already running in the background gorotutine so not blocking for the main flow.